ifrs17/Utils/Queries.ipynb

```diff
@@ -217,14 +217,26 @@
         {
             "cell_type": "code",
             "source": [
-                "public static async Task<Dictionary<string, Dictionary<int, YieldCurve>>> LoadCurrentYieldCurveAsync(this IQuerySource querySource, Args args, ",
-                "\n                                                                                                      IEnumerable<DataNodeData> dataNodes)",
-                "\n{",
-                "\n    var contractualCurrenciesInScope = dataNodes.Select(dn => dn.ContractualCurrency).ToHashSet();",
-                "\n    return (await querySource.LoadCurrentAndPreviousParameterAsync<YieldCurve>(args, ",
+                "public static async Task<Dictionary<string, Dictionary<int, YieldCurve>>> LoadCurrentYieldCurveAsync(this IQuerySource querySource, Args args,",
+                "\n                                                                                                                                IEnumerable<DataNodeData> dataNodes)",
+                "\n{    ",
+                "\n    var currentYieldCurveByGoc = new Dictionary<string,  Dictionary<int, YieldCurve>>();",
+                "\n    ",
+                "\n    foreach (var dn in dataNodes)",
+                "\n    {",
+                "\n        var loadedYc = await querySource.LoadCurrentAndPreviousParameterAsync<YieldCurve>(args, ",
                 "\n                                                                               x => x.Currency,",
-                "\n                                                                               x => contractualCurrenciesInScope.Contains(x.Currency)",
-                "\n                                                                               ));",
+                "\n                                                                               x => x.Currency == dn.ContractualCurrency ",
+                "\n                                                                                    && (dn.ValuationApproach == ValuationApproaches.VFA ",
```

Comment:
I don't understand this part. The current curve is always used, why this?

Reply:
Sara: It is always used, but the additional filter in the query for YieldCurveName happens only for VFA

Reply:
In connection with the name above, I might now understand.
I don't see why it is related to VFA though... if there is a name in the dn, you should take the curve with the name, right?
Basically you are saying to ignore the name if the valuation approach is not VFA.
Since we don't have VFA, I would not add this logic, we can think later.
I remember talking to @dcolleoni about this case, that I think says for the current curve always ignore the name.

Reply:
I would leave it now as it is done already...